src/core/node/object-node.ts

```diff
@@ -215,7 +215,9 @@ export class ObjectNode<C, S, T> extends BaseNode<C, S, T> {
     const type = this.type
 
     try {
-      this.storedValue = type.createNewInstance(this._childNodes)
```

Comment:
I still want to fix this, but I've got this error on `master`, so it may not be because of the TS upgrade

<img width="971" alt="Screenshot 2024-02-12 at 8 12 22 PM" src="https://github.com/mobxjs/mobx-state-tree/assets/29899554/b9494de6-baf2-4167-b9dc-d2832d60d338">